draft-tjjk-cared.md

```diff
@@ -1,336 +1,405 @@
+specified by the operator of the encrypted DNS server.
+
+For example, an encrypted DNS server that is configured to allow
+some clients to resolve certain names while other clients are not
+allowed needs to identify the client so that the resolution behaviour
+for those names can be implemented accurately.
+
+Encrypted DNS servers SHOULD NOT attempt to authenticate clients
+to identify the appropriate resolution policy to use when the
+difference in resolution behavior between them is not imposed by
+the operator of the server but instead is chosen by the client.
+
+For example, some public encrypted DNS services provide clients
+with the option of blocking resolution requests for particular
+categories of domain names, such as names associated with malware
+distribution, adult content or advertisers, and makes servers with
+particular resolution policies available on different IP addresses.
+A client who wishes to opt-in to a server with a particular resolution
+policy can do so by sending queries to the corresponding IP address,
+and no client identification is required.
+
+## When clients should attempt to authenticate
+
+An encrypted DNS client MUST NOT offer authentication to any encrypted
+DNS server unless it was specifically configured to expect that
+server to require authentication, independent of the mechanism by
+which the client chose or discovered the encrypted DNS server to
+use.
+
+An encrypted DNS client MUST NOT offer authentication when the
+server connection was established using DDR {{?RFC9462}}, even if
+the IP address of the original DNS server was specifically configured
+for the encrypted DNS client as one that might require authentication.
+This is because in that circumstance there is not a pre-existing
+relationship with the encrypted DNS server (or else DDR bootstrapping
+into encrypted DNS would not have been necessary).
+
+TBD: what to do with EDSR destinations {{EDSR-REDIRECTION}}
+
+An encrypted DNS client MAY choose to present authentication to a
+server that requests it, but is not required to do so; for example,
+a client MAY choose instead not to use the server. If a client does
+present an identity to a server, the identity SHOULD be unique to
+that server, to reduce the risk of a client's resolution history
+on multiple colluding servers being correlated.
+
+# Recommendations
+
+The following requirements were considered when formulating the
+recommended authentication mechanism for encrypted DNS clients.
+The authentication mechanism:
+
+# SHOULD be per-connection, not per-query, e.g. to
+avoid unnecessary payload overheads
+# SHOULD use open, standard mechanisms where
+possible, e.g. to avoid vendor lock-in and specialized cryptography
+# SHOULD be reusable across multiple encrypted DNS protocols, e.g.
+to avoid protocol preference
+# SHOULD NOT require human user interaction to complete authentication
+
+This document concludes that the best mechanism for encrypted
```

Comment:
Same problem here: "#" is not a bullet, yet fmt(1) seems to have made it one? Please change this back; the markdown is parsed hideously now (and does not preserve the next line as being part of the same header).